### PR TITLE
Add documentation on error handling, specify missing header in error message

### PR DESCRIPTION
**Pull request by daliwali**
Note: in my opinion it should not throw as many errors as it does, allow it to return null values and then it is up to the programmer to check if it succeeded. This is particularly annoying in the case when a request does not have `Content-Type` specified.

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+coverage/
+.travis.yml
```
Comment (dougwilson) [inline .npmignore]: Remove this file, please; `package.json` already contains this information.

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "asi": true,
+  "boss": true,
+  "laxbreak": true,
+  "unused": true,
+  "undef": true,
+  "node": true
+}
```
Comment (dougwilson) [inline .jshintrc]: No jshint for our projects. we may start using `standard`, but IDK.

**Comment by dougwilson** [comment]
> This is particularly annoying in the case when a request does not have Content-Type specified.

Gotcha. We should probably add a content-type getter function which returns the content-type or `undefined` if it's not there.

**Comment by rlidwka** [comment]
Please remove dotfiles because:

1. It should be a separate pull request
2. They should be added through [this](https://github.com/jshttp/style-guide) repository first
3. `.npmignore` is completely useless because this project uses `files` property
4. Adding `.jshintrc` was [discussed](https://github.com/jshttp/style-guide/issues/8) before and decided against

**Comment by dougwilson** [comment]
P.S. @Fishrock123 , can you update the labels for this repo :)?

**Comment by daliwali** [comment]
@dougwilson if a key on an object is missing, it already returns `undefined` and the rest of the function will still throw an error anyways. I just don't want to have to wrap this in a try...catch block because of a missing header :(

**Comment by dougwilson** [comment]
@daliwali I know. I'm talking about making a new function exported by this module that _doesn't do that_ :)

**Comment by dougwilson** [comment]
@daliwali how does this API look:
```js
var contentType = require('content-type')
var type = contentType(req)

// type is a parsed type or undefined
```

**Comment by daliwali** [comment]
@dougwilson I like that, very concise.

**Comment by dougwilson** [comment]
P.S. @Fishrock123 , can you update the labels for this repo :)?

**Comment by dougwilson** [comment]
@daliwali sorry to bug you again :) Bu for the `contentType(req)` API, in your opinion, what should it d if there is a `Content-Type` header, but it's invalid?

**Comment by daliwali** [comment]
@dougwilson Ah, perhaps it really should throw an error. The alternative is to fail silently, which wouldn't feel good either. Throwing an error would feel consistent with some other parsing methods, like `JSON.parse`. Ok I take it back, errors are fine all around. It was just disorienting to see `argument string is required to be a string` when I passed in a request object.

**Comment by dougwilson** [comment]
Ok, we can think on it some more. For now, I'm' going to release your changes from this PR; feel free to open a new issue for anything ;)

**Pull request closed without merging**